core/renderers/common/path_object.ts

```diff
@@ -268,37 +268,33 @@ export class PathObject implements IPathObject {
     connectionPath: string,
     offset: Coordinate,
     rtl: boolean,
-  ) {
-    if (this.connectionHighlights.has(connection)) {
-      if (this.currentHighlightMatchesNew(connection, connectionPath, offset)) {
-        return;
-      }
-      this.removeConnectionHighlight(connection);
+  ): SVGElement {
+    const transformation =
+      `translate(${offset.x}, ${offset.y})` + (rtl ? ' scale(-1 1)' : '');
+
+    const previousHighlight = this.connectionHighlights.get(connection);
+    if (previousHighlight) {
+      // Since a connection already exists, make sure that its path and
+      // transform are correct.
+      previousHighlight.setAttribute('d', connectionPath);
+      previousHighlight.setAttribute('transform', transformation);
```

Comment:
The ones highlighted in this comment:

https://github.com/google/blockly/blob/896dbd7c11e532661fc6ab75ad759b604bbf38fb/core/renderers/common/path_object.ts#L279-L280

I.e. if we don't recompute the transform for an existing connection highlight, they still seem to magically go to the right place even if the block moves.

Reply:
Is it across block movements or blocks changing shape?
If it's about block movements, that's not surprising, because the transform is relative to the origin of the block's SVG and not to the rest of the page. Translating the block therefore also translates the connection highlights and all of the attached blocks.

Reply:
Sorry--not at all sure why the comment started a new thread.

I think it might be the relative behavior, then. I tried a few box shape changing things and it didn't make a difference, but I probably didn't get the block to resize in the correct way to trigger a failure (since the connection highlight doesn't span the whole blog length like the marker does, so simply changing the width isn't quite sufficient to produce the case that would break the connection highlight w/o this update logic).

Reply:
Discussed over chat, but it largely seems that I may have missed cases where connection actually changes size (and am, admittedly, still slightly confused on why those cases _seem_ to work without this code in place). Nevertheless, it's definitely more correct to include this so continuing with that plan.